resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -430,6 +433,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
       .endMetadata()
       .withNewSpec()
         .withHostname(hostname)
+        .withNodeSelector(nodeSelector.asJava)
```

Comment:
@ash211 This `node selector` is not same as `node affinity` that I referred to as 1.6+ featured. As @foxish mentioned, `node affinity` is a superset of `node selector`. K8s added `node affinity` later to support more general use cases. From the reference [doc](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/):
> nodeSelector provides a very simple way to constrain pods to nodes with particular labels. The affinity/anti-affinity feature, currently in beta, greatly expands the types of constraints you can express.
...
> nodeSelector continues to work as usual, but will eventually be deprecated, as node affinity can express everything that nodeSelector can express